src/v3/src/types/schema.ts

```diff
@@ -271,6 +271,11 @@ export interface FieldElement extends UISchemaElement {
    * This only applies for profile enrollment view.
    */
   showAsterisk?: boolean;
+  /**
+   * @description Determines if the app should add a hyphen after the field's 4th character
+   * This only applies for device code view.
+  */
+  hyphenate?: boolean;
```

Comment:
I'm concerned about the number of view-specific properties we're adding to elements. Can we explore making this more generic and then allow the transformer to pass in the very specific function to execute after the key input is pressed.

Reply:
Curious if in this case we can leverage form field validation in `handleChange` and `handleBlue` to do this. If so it would also solve the issue above mentioned in this code comment:

> // NOTE: Hyphenation is triggered by a KeyboardEvent and only occurs upon typing, not pre-filling

@glenfannin-okta are we able to modify the field data from the validator functions right now in a reliable way? If not, is it just a matter of adding another property to the validator function callback? If so, I'm in favor of extending the validator's functionality to also rewrite the current value of the field so long as it isn't terribly complex or causes weird looping issues.

Reply:
Would making this change to use form field validation provide value to any other flows? 

If not, I feel like the changes required to add this form field validation might not be worth it, especially because other OTP inputs don't use hyphenation, so it might actually be more consistent to have device code match this behavior since it's also essentially an OTP? 

It already seemed a bit odd to me that the device code is hyphenated purely for UX reasons, but other OTPs are not.

Reply:
@shawyu-okta It may be worth a try, I think a new hook would need to be created, then use that hook within `src/v3/src/components/hocs/withFormValidationState.tsx` and either add it to the `handleChange` function or create a new property (for keyup) and add it to the keyup attribute of the input. 

@trevoring-okta Wanna give this a try, I can assist with it.